### PR TITLE
Decrease supermatter delam rate in vacuum

### PR DESCRIPTION
**Pull request by Talkarcabbage**
Added a tweakable multiplier for situations where the supermatter is in no atmosphere (primarily, when someone opens the core vent). Right now, it delaminates in around ten seconds or less from its first announcement (especially if emitter is on) with no chance for engineering to respond. It should take much longer to do so now than before. 

In testing, the speed was 4-8% per announcement on a reasonable number of shots (by aurora standards). It will still delaminate, including in space tiles (via core eject, etc, and still assuming it's been shot) but the rate will be less... instant. Turning the emitter on will still naturally increase the rate of damage as it fires.

Also decreased the damage cap some for in-atmosphere damage, also for being able to have time to respond to delaminations (said cap still scales with emitter shots)

### CODE REVIEW

File: code/modules/supermatter/supermatter.dm [hunk 1]
```diff
@@ -20,8 +20,8 @@
 #define DECAY_FACTOR 700			//Affects how fast the supermatter power decays
 #define CRITICAL_TEMPERATURE 5000	//K
 #define CHARGING_FACTOR 0.05
-#define DAMAGE_RATE_LIMIT 3			//damage rate cap at power = 300, scales linearly with power
-
+#define DAMAGE_RATE_LIMIT 2			//damage rate cap at power = 300, scales linearly with power
+#define SPACED_DAMAGE_FACTOR 0.1	//multiplier for damage taken in a vacuum, but on a tile. Used to prevent/configure near-instant explosions when vented
```
Comment (Sindorman) [inline code/modules/supermatter/supermatter.dm]: so this reduces vacuum damage by 90% according to formula.

Comment (Sindorman) [reply code/modules/supermatter/supermatter.dm]: because `number * 0.1 = 10% of the number`

Comment (Talkarcabbage) [reply code/modules/supermatter/supermatter.dm]: Yes. Added it as a configurable number in case anyone thinks it should be higher or lower.

**Comment by Geevies** [comment]
!review

**Comment by NonQueueingMatt** [comment]
@Talkarcabbage This needs a changelog.

**Comment by Talkarcabbage** [comment]
So I would do something like this?
`html/changelogs/sm_delam_nerf.yml`
->
```
author: Talkarcabbage

delete-after: True

changes: 
  - tweak: Substantially decreased supermatter delamination rate in complete vacuum.```

**Comment by NonQueueingMatt** [comment]
Yes

**Comment by Talkarcabbage** [comment]
Following up after a few ages... For matt and (especially) hayden's feedback, one of the big situations that this nerf is directed towards is situations where the vent is opened and the emitter is turned on simultaneously and then left alone. While the engine still has a tiny bit of coolant flowing in from the pipes or is making some on its own, it maintains zero damage intake while eating up emitter shots, especially (but not limited to) phoron coolant. 

Without the nerf, when it hits vacuum, it suddenly delaminates, usually in less than 15-30 seconds (observed on an engine that was running on only six shots, with the 1-2 phoron setup most people are using): 
https://puu.sh/EMGUz/1d85e09ff6.png
(quick enough to only have announcements at 86%, 23%, then 0%)

The emitter sounds going through walls actually changed my opinion on this a bit, since it makes it easier to catch this particular situation before complete disaster. A 0.5 instead of 0.1 factor should give enough time for an AI or an immediately-nearby responder (if they didn't hear the emitter until delamination started) time to open the emergency valves or eject the SM if they're quick, but not much more time than that. It _should_ still end up being a "respond immediately or die" situation.

@NonQueueingMatt I can bump it up higher than 0.5 if you still think it should be. It's a _very_ quick way to end a round on live right now, but it's been a few weeks since I've seen it done at all.

File: code/modules/supermatter/supermatter.dm [hunk 2]
```diff
@@ -20,8 +20,8 @@
 #define DECAY_FACTOR 700			//Affects how fast the supermatter power decays
 #define CRITICAL_TEMPERATURE 5000	//K
 #define CHARGING_FACTOR 0.05
-#define DAMAGE_RATE_LIMIT 3			//damage rate cap at power = 300, scales linearly with power
-
+#define DAMAGE_RATE_LIMIT 2			//damage rate cap at power = 300, scales linearly with power
+#define SPACED_DAMAGE_FACTOR 0.5	//multiplier for damage taken in a vacuum, but on a tile. Used to prevent/configure near-instant explosions when vented
```
Comment (NonQueueingMatt) [inline code/modules/supermatter/supermatter.dm]: Undefine this at the end of the file like so:

`#undef SPACED_DAMAGE_FACTOR`

Comment (Talkarcabbage) [reply code/modules/supermatter/supermatter.dm]: Guessing that keeps it from being visible in other places?

Comment (NonQueueingMatt) [reply code/modules/supermatter/supermatter.dm]: Correct

**Pull request merged**